sympy/functions/special/bessel.py

```diff
@@ -1124,9 +1124,11 @@ def fdiff(self, argindex=1):
         else:
             raise ArgumentIndexError(self, argindex)
 
-    @staticmethod
+
     @cacheit
-    def taylor_term(n, x, *previous_terms):
+    def taylor_term(self, n, x, *previous_terms):
+        if self.args[0] != x:
+            return super(sin, self).taylor_term(n, x, *previous_terms)
```

Comment:
This is in Airy function class. Why `super(sin, self)`? And why the change in Airy but not any other functions?